types/prosemirror-commands/index.d.ts

```diff
@@ -4,13 +4,26 @@
 //                 David Hahn <https://github.com/davidka>
 //                 Tim Baumann <https://github.com/timjb>
 //                 Patrick Simmelbauer <https://github.com/patsimm>
+//                 Mike Morearty <https://github.com/mmorearty>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 3.0
 
 import { MarkType, Node as ProsemirrorNode, NodeType, Schema } from 'prosemirror-model';
 import { EditorState, Transaction } from 'prosemirror-state';
 import { EditorView } from 'prosemirror-view';
 
+export interface Command<S extends Schema = any> {
+  (
+    state: EditorState<S>,
+    dispatch: (tr: Transaction<S>) => void,
+    view: EditorView<S>
+  ): boolean;
+}
+
+export interface Keymap<S extends Schema = any> {
+  [key: string]: Command<S>;
+}
```

Comment:
It's not really clear to me if these types should go here, or in prosemirror-keymap. They are used by both.

Reply:
I think that prosemirror-commands is the best place for the `Command` interface. Unfortunately, moving the `Keymap` interface to prosemirror-keymap would introduce a cycle. Thus, the current solution seems to be the best one to me.